armory/scenarios/poison.py

```diff
@@ -508,28 +506,119 @@ def evaluate_current(self):
         if self.use_poison:
             self.run_attack()
 
-    def finalize_results(self):
-        if hasattr(self, "fairness_metrics") and not self.check_run:
-            self.test_set_class_labels = sorted(self.test_set_class_labels)
-            if self.train_set_class_labels != self.test_set_class_labels:
-                log.warning(
-                    "Test set contains a strict subset of train set classes. "
-                    "Some metrics for missing classes may not be computed."
-                )
-            self.fairness_metrics.add_cluster_metrics(
-                self.x_poison,
-                self.y_poison,
-                self.poison_index,
-                self.indices_to_keep,
-                # Get unpoisoned test set
-                config_loading.load_dataset(
-                    self.config["dataset"],
-                    split="test",
-                    num_batches=None,
-                    **self.dataset_kwargs,
-                ),
+        if self.num_export_batches > self.sample_exporter.saved_batches:
```

Comment:
looks like lucas's code removed this block (through line 517).  self.sample_exporter has no attribute 'saved_batches'

Reply:
But when I remove it (locally) and run, it exports all the images, not just up to export_batches from the config.  so there's something else missing

Reply:
@lcadalzo may be able to find the issue faster than me, I'm not yet very familiar with the new exporting code

Reply:
@swsuggs after removing this block and running with `"export_batches": 2`, my output looks right:

```
I have no name!@d16b7467e2a5:/workspace$ ls /armory/outputs/2022-05-26T191142.588735/saved_samples_1653592697.6484163/
batch_0_ex_0_scenario.x.png  batch_0_ex_0_scenario.x_adv.png  batch_1_ex_0_scenario.x.png  batch_1_ex_0_scenario.x_adv.png  predictions.pkl
```

Which config are you using? What did you set `"export_batches"` to?

Reply:
I also changed batch_size to 1

Reply:
It was my error.  I ran a gtsrb dlbd config, but I was thinking of witches' brew which automatically uses a test batch_size of 1.  My batch size was still 512 after all.

Reply:
Curious what config you used though.  When I set batch_size to 1 for gtsrb dlbd, I get an error:
`ValueError: Expected more than 1 value per channel when training, got input size torch.Size([1, 300])`

Reply:
I was also using gtsrb dlbd, but for speed's sake I commented out the `fit()` call in the scenario code. When I uncomment it, I see the error. Weird bc the error goes away if I set batch_size to 2. I'll create an issue for it